### DISCUSSION TITLE
Advanced Planner / Vector Search Capabilities - What is Product Strategy?

### DISCUSSION THREAD

**Discussion by cchighman** [microsoft/semantic-kernel#1327] [Q&A] [answered]
Looking for guidance around future plans for a handful of capabilities and what the stance is on contribution as it relates to advanced planner / vector capabilities.  We would love to contribute back with respective capabilities.

**Suppose you have an ask which contains multiple questions and you're using embeddings to search a vector database**
Under these circumstances, it seems you may need to be able to break down the overall question into steps that  may involve a cycle of getting vector matches for first question, asking GPT to formulate an appropriate second vector search in order to get matches for the second question, etc.

This process described above seems like it would be an important skill for Planner.   Or at least ability to derive multiple intents from an utterance along with any associated slot values.  I noticed in the samples, say the QASkill for example, are not included in the core product.  Is it thought there will be an ecosystem of skills or otherwise that are essentially community-sourced extensions that wouldn't be appropriate in the product unless they represented some core building block activity?  What are thoughts here?

**Vector Search Skills** 
Are there any thoughts to offering capabilities to do processing activities on stored vectors that might seek to build new vectors from existing and perhaps summarize, etc, as part of a general strategy around improving "sharpness" of memory or is it thought this type of activity would be outside the scope of Semantic Kernel with some other process doing this. I only ask because of the inherent vector skills in the core of the product.

Thanks!

**Comment by markwallace-microsoft** [comment, marked as answer]
This ADR describes our memory strategy: https://github.com/microsoft/semantic-kernel/blob/a24762d600e1de787bd9292c08abde41129fe8fa/docs/decisions/0072-agents-with-memory.md
